torchtune/datasets/_alpaca.py

```diff
@@ -6,20 +6,90 @@
 
 from functools import partial
 
-from torchtune.datasets._instruct import instruct_dataset, InstructDataset
+from typing import Any, Dict, Mapping, Optional, Union
 
-from torchtune.modules.tokenizers import ModelTokenizer
+from torchtune.data import Message
+from torchtune.datasets._packed import PackedDataset
+from torchtune.datasets._sft import SFTDataset
+from torchtune.modules.transforms import Transform
+
+
+class AlpacaToMessages(Transform):
```

Comment:
This I find a bit confusing. Why are most message transforms in `_messages.py` but then this one is here? (I know there is some logic custom to Alpaca, but still)

Reply:
the others are generic that can apply to any dataset of similar structure. Alpaca's is specific to alpaca and should remain here. kind of like how model specific components are in model folders but all generic components go on modules